numpy/_core/numeric.pyi

```diff
@@ -658,3 +658,9 @@ def isclose(
 def array_equal(a1: ArrayLike, a2: ArrayLike, equal_nan: bool = ...) -> bool: ...
 
 def array_equiv(a1: ArrayLike, a2: ArrayLike) -> bool: ...
+
+def astype(
+    x: ArrayLike,
+    dtype: DTypeLike,
+    copy: bool = ...,
+) -> NDArray[Any]: ...
```

Comment:
`astype` could really use an overload here for all `generic`-parametrizable dtype-likes, as we can quite easily statically express the dtype of the returned arrays in those cases.
```suggestion
@overload
def astype(
    x: ArrayLike,
    dtype: _DTypeLike[_SCT],
    copy: bool = ...,
) -> NDArray[_SCT]: ...
@overload
def astype(
    x: ArrayLike,
    dtype: DTypeLike,
    copy: bool = ...,
) -> NDArray[Any]: ...
```

Reply:
I committed it - thanks!